### PR TITLE
⚡️(circle) split bootstrap tests

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

When a running job has no output since a certain amount of time (10 minutes by default), CircleCI kills this job. As the `edxapp` application bootstrapping can be longer than that, this prevents us from testing it.

## Proposal

- [x] ~Increase this limit to 30 minutes (yes, we can do that :tada:)~ [1]  **edit**: this does not work
- [x] add the `bin/ci-bootstrap` script
- [x] split the `test-boostrap` job in multiple jobs - #91 

[1] https://support.circleci.com/hc/en-us/articles/360007188574-Build-has-hit-timeout-limit

### CODE REVIEW

**Review by sampaccoud — APPROVED** [review]

**Comment by jmaupetit** [comment]
Looks like the `no_output_timout` parameter does nothing.

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,6 +32,38 @@ aliases:
...
 
+  - &install_oc
+    run:
+      name: Install the OC client
+      command: |
+        wget https://github.com/openshift/origin/releases/download/v3.9.0/openshift-origin-client-tools-v3.9.0-191fece-linux-64bit.tar.gz -P /tmp/openshift/
+        tar xzf /tmp/openshift/openshift*.tar.gz --strip-components=1 -C /tmp/openshift/
+        cp /tmp/openshift/oc $HOME/bin/
+
+  - &configure_oc
```
Comment (sampaccoud) [inline .circleci/config.yml]: configure_os or configure_openshift_cluster?

File: .circleci/config.yml [hunk 2]
```diff
@@ -32,6 +32,38 @@ aliases:
...
+    run:
+      name: Configure Docker insecure registries for OpenShift cluster
```
Comment (sampaccoud) [inline .circleci/config.yml]: Change the name now that you are doing insecure registries but also ES?

File: .circleci/config.yml [hunk 3]
```diff
@@ -32,6 +32,38 @@ aliases:
...
+      command: |
+        # Elasticsearch requires to increase this setting's default value
+        sudo sysctl -w vm/max_map_count=262144
+        sudo bash -c "echo '{\"insecure-registries\": [\"172.30.0.0/16\"]}' > /etc/docker/daemon.json"
+        sudo service docker restart
+
+  - &run_oc
```
Comment (sampaccoud) [inline .circleci/config.yml]: run_os or run_openshift_cluster?

File: apps/edxapp/templates/cms/_configs/settings.yml.j2 [hunk 4]
```diff
@@ -6,3 +6,14 @@ SITE_NAME: "{{ project_display_name }}"
 
 LMS_BASE: "{{ edxapp_lms_host }}"
 CMS_BASE: "{{ edxapp_cms_host }}"
+
+# Celery Broker
+# FIXME: we should use the redis_app_* default vars
```
Comment (sampaccoud) [inline apps/edxapp/templates/cms/_configs/settings.yml.j2:11]: I did not understand this sentence?

File: bin/ci-bootstrap [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+# usage: ci-bootstrap [APP]
+
+# First script argument is the application's name. It defaults to
+# 'hello'
+app="${1:-hello}"
+
+# Run the bootstrap.yml playbook with the ansible vault password
+# passed to stdin
+echo "arnold" | docker run --rm -i \
+  --env-file env.d/development \
```
Comment (sampaccoud) [inline bin/ci-bootstrap]: have a ci env file?

**Review by sampaccoud — APPROVED** [review]
Just naming suggestions but ok to merge.

File: apps/edxapp/templates/cms/_configs/settings.yml.j2 (hunk 4 above)
Comment (jmaupetit) [reply apps/edxapp/templates/cms/_configs/settings.yml.j2:11]: The redis app should be bundled with configuration variables (image, tag, host, port, ...). We will use those variables when they will be available (from your PR).

File: .circleci/config.yml (hunk 3 above)
Comment (jmaupetit) [reply .circleci/config.yml]: `run_openshift_cluster`

File: .circleci/config.yml (hunk 1 above)
Comment (jmaupetit) [reply .circleci/config.yml]: `configure_openshift_cluster`

File: apps/edxapp/templates/cms/_configs/settings.yml.j2 (hunk 4 above)
Comment (sampaccoud) [reply apps/edxapp/templates/cms/_configs/settings.yml.j2:11]: IC :+1:

**Pull request merged**